### PR TITLE
Fixes issue #384 by adding an noProxy configuration parameter npm execution

### PR DESCRIPTION
**Pull request by matthewhubblerose**
Fixes issue #384.

Also moved some logging around to make things less confusing.

### CODE REVIEW

**Comment by eirslett** [comment]
Maybe rename "noProxy" to "npmInheritsProxyConfigFromMaven", which defaults to true? It's a much longer property name, but also more descriptive. npm could still be configured with a proxy, even if this setting is configured.

**Comment by matthewhubble** [comment]
Agreed - that's a much better name.

**Comment by eirslett** [comment]
Looks like the npm dependencies in the example project/integration tests are out of date, I guess that's why the build is failing :-(

**Comment by matthewhubble** [comment]
Looks like a problem with phantomjs. Wierd - as the tests pass a couple of hours ago with my first commit

**Comment by matthewhubble** [comment]
```
[INFO] [ERROR] Error requesting archive.
[INFO] [ERROR] Status: 429
[INFO] [ERROR] Request options: {
[INFO] [ERROR]   "uri": "https://bitbucket.org/ariya/phantomjs/downloads/phantomjs-1.9.8-macosx.zip",
```

**Comment by matthewhubble** [comment]
Is it possible to re-run the checks? Perhaps it's was some temporary error - I managed to download from that URL no problem

**Comment by matthewhubble** [comment]
It's a problem with the phantom-js download from:
https://bitbucket.org/ariya/phantomjs/downloads/phantomjs-1.9.8-windows.zip

When I hit that URL, I get a response with this text:
Rate limit for this resource has been exceeded

Status code is 429 = Too Many Requests.

**Comment by eirslett** [comment]
Looks like it worked now!

File: frontend-maven-plugin/src/main/java/com/github/eirslett/maven/plugins/frontend/mojo/NpmMojo.java [hunk 1]
```diff
@@ -12,6 +12,7 @@
 import org.sonatype.plexus.build.incremental.BuildContext;
 
 import java.io.File;
+import java.util.ArrayList;
```
Comment (eirslett) [inline frontend-maven-plugin/src/main/java/com/github/eirslett/maven/plugins/frontend/mojo/NpmMojo.java]: Where is this used?

**Comment by eirslett** [comment]
Looks good! It might also be worthwhile to add a short notice in README.md about proxies, and then link to the wiki with a longer article that explains this functionaliry?

**Comment by matthewhubble** [comment]
I've added the documentation to the README. I think it explains everything there, I don't think the Wiki will also need updating.

File: frontend-maven-plugin/src/main/java/com/github/eirslett/maven/plugins/frontend/mojo/NpmMojo.java (hunk 1 above)
Comment (matthewhubble) [inline frontend-maven-plugin/src/main/java/com/github/eirslett/maven/plugins/frontend/mojo/NpmMojo.java]: It was being used to create an empty list. I've now changed it to use `Collections.emptyList` to be more consistent with the rest of the code.

**Comment by eirslett** [comment]
Thanks a lot! Could you please squash the commits into one? And then I'll merge it :-)

**Pull request closed without merging**